runtime/codegen/registry.go

```diff
@@ -137,6 +136,33 @@ func (r *registry) find(path string) (*Registration, bool) {
 	return reg, ok
 }
 
+// Config calls the WithConfig.Config method on the provided value and returns
+// the result. If the provided value doesn't have a WithConfig.Config method,
+// Config returns nil.
+//
+// Config panics if the provided value is not a pointer to a struct.
+func Config(v reflect.Value) any {
```

Comment:
I'm getting bothered about this being an exported function that isn't called from the generated code.

Would it be possible to place this in Registration.ConfigFn? codegen.Register() can auto-fill it when called.

Not a big deal if too painful.

Reply:
I moved this function to an internal `config` package. I also added a TODO to move the `ComponentConfigValidator` function right below it as well. It's also not used by the generated code but moving it is a bit trickier.